crates/aggregate_builder/src/lib.rs

```diff
@@ -0,0 +1,749 @@
+    }
+}
+
+impl Parse for AggregateParallelSafe {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        let _: Token![const] = input.parse()?;
+        let name: syn::Ident = input.parse()?;
+        if name != "PARALLEL_SAFE" {
+            error!(name.span(), "unexpected const `{}` expected `PARALLEL_SAFE`", name)
+        }
+        let _: Token![:] = input.parse()?;
+        let ty: syn::Ident = input.parse()?;
+        if ty != "bool" {
+            error!(ty.span(), "unexpected type `{}` expected `bool`", ty)
+        }
+        let _: Token![=] = input.parse()?;
+        let value = input.parse()?;
+        let _: Token![;] = input.parse()?;
+        Ok(Self { value })
+    }
+}
+
+impl Parse for AggregateFn {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        let mut attributes = input.call(syn::Attribute::parse_outer)?;
+        let _: Token![fn] = input.parse()?;
+        let ident = input.parse()?;
+
+        let contents;
+        let parens = syn::parenthesized!(contents in input);
+
+        let mut args = Punctuated::new();
+        while !contents.is_empty() {
+            let arg = contents.parse()?;
+            args.push(arg);
+            if contents.is_empty() {
+                break
+            }
+            let comma: Token![,] = contents.parse()?;
+            args.push_punct(comma);
+        };
+
+        let ret = input.parse()?;
+        let body = input.parse()?;
+
+        let expected_path = parse_quote!(sql_name);
+        let sql_name = match take_attr(&mut attributes, &expected_path) {
+            None => None,
+            Some(attribute) => attribute.parse_args()?
+        };
+        if !attributes.is_empty() {
+            error!(attributes[0].span(), "unexpected attribute")
+        }
+        Ok(Self {
+            ident,
+            sql_name,
+            parens,
+            args,
+            ret,
+            body,
+        })
+    }
+}
+
+impl Parse for AggregateArg {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        let arg: syn::FnArg = input.parse()?;
+        let mut rust = match arg {
+            syn::FnArg::Typed(pat) => pat,
+            _ => error!(arg.span(), "`self` is not a valid aggregate argument"),
+        };
+        let sql = {
+            let expected_path = parse_quote!(sql_type);
+            let attribute = take_attr(&mut rust.attrs, &expected_path);
+            match attribute {
+                None => None,
+                Some(attribute) => attribute.parse_args()?
+            }
+        };
+        Ok(Self {
+            rust,
+            sql,
+        })
+    }
+}
+
+fn take_attr(attrs: &mut Vec<syn::Attribute>, path: &syn::Path)
+-> Option<syn::Attribute> {
+    let idx = attrs.iter().enumerate().find(|(_, a)| &a.path == path);
+    match idx {
+        None => None,
+        Some((idx, _)) => {
+            let attribute = attrs.remove(idx);
+            Some(attribute)
+        }
+    }
+}
+
+//
```

Comment:
Bit of cleanup needed here.

Reply:
That's entirely deliberate: it's there to serve as a visual marker when scrolling looking for the expansion part of the proc-macro. I'll turn it into a real comment.